packages/xo-web/src/xo-app/home/index.js

```diff
@@ -480,22 +480,23 @@ export default class Home extends Component {
     selectedItems: {},
   }
 
-  get page() {
-    return this.state.page
-  }
-  set page(activePage) {
-    this.setState({ activePage })
-  }
-
   componentWillMount() {
     this._initFilterAndSortBy(this.props)
   }
 
   componentWillReceiveProps(props) {
+    const {
+      location: { query },
+      type,
+    } = props
+
+    if (+query.p !== this.state.activePage) {
```

Comment:
What if there is no `p` in query or if it's not a valid number?

How is initialized `activePage`?

Reply:
`activePage` is not initialized before we click on `Pagination` component or we set it on `componentWillReceiveProps`. But in [Pagination](https://github.com/vatesfr/xen-orchestra/blob/71088dd49a60439797ff4ca92663f2a7bb343512/packages/xo-web/src/xo-app/home/index.js#L1207), we have a default value when `activePage` is undefined.
I will set it in `state`.

If there is no `p` in query, because we have a default value for activePage, we don't have bug. But if the number is not valid, we have no data. I will fix it.

Reply:
Yes, if `p` does not exist or is not a valid number, it will defined `activePage` at `NaN`.